platform-sdk/swirlds-pairings-api/src/main/java/com/swirlds/pairings/api/GroupElementAggregator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.pairings.api;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Aggregator for group elements
+ * <p>
+ * Exists to take advantage of batch operations in the underlying group
+ */
+public class GroupElementAggregator {
+    private final List<GroupElement> elements;
+
+    /**
+     * Constructor
+     */
+    public GroupElementAggregator() {
+        this.elements = new ArrayList<>();
+    }
```

Comment:
Does it make sense to create an empty aggregation? Do you think we can remove the option from the API?
A static builder that receives the first element as a parameter.
But also, I might be off and have to look at how you are using it.

Reply:
The problem here is that you'd need to either create an identity element as your first element (and pay the corresponding performance price).

Or alternatively mangle loops that aggregate many elements with special first-iteration logic